app/init.php

```diff
@@ -1219,8 +1221,12 @@ function (mixed $value) {
             $user = $dbForProject->getDocument('users', $jwtUserId);
         }
 
-        if (empty($user->find('$id', $jwtSessionId, 'sessions'))) { // Match JWT to active token
+        $session = $user->find('$id', $jwtSessionId, 'sessions');
+
+        if (empty($session) || Auth::isSessionExpired($session)) { // Match JWT to active token
```

Comment:
@stnguyen90 This missing time check was the cause for this #8000 issue. Therefore added this session expired check over here. That should be also fixed now, as per my tests. 🤞 

Just FYI - For JWT we still don't have a secret check like we have in sessions, that's because we don't have that data in JWT. But I think that's the point of directly storing the userID, so that we don't have to depend on the session secret, right?

Reply:
> Just FYI - For JWT we still don't have a secret check like we have in sessions, that's because we don't have that data in JWT. But I think that's the point of directly storing the userID, so that we don't have to depend on the session secret, right?

Yes. We just need to check the JWT is valid (`$jwt->decode()`) and if it is, trust the value is valid.